docs/migrations/v1_to_v2.md

```diff
@@ -0,0 +1,55 @@
+- Refactored our APIs for an easier and safer use.
+
+For a more complete overview, please refer to our [release notes on GitHub](https://github.com/DataDog/dd-trace-dotnet/releases).
+
+## Upgrading from 1.x to 2.0
+
+### Code changes
+
+Most of our api changes do not require any changes to your code, but some patterns are no longer supported or recommended. Please refer to [Datadog.Trace documentation](https://github.com/DataDog/dd-trace-dotnet/tree/v2.0.1/docs/Datadog.Trace#upgrading-from-1x-to-20) for more information.
+
+### What if you are relying on .NET Framework lower than 4.6.1
+
+If you are currently targeting version < 4.6.1, we suggest you upgrade in line with [Microsoft's guidance](https://docs.microsoft.com/en-us/lifecycle/products/microsoft-net-framework). On our end, we will support fixing major bugs on top of Tracer v1.31.
+
+That said, automatic instrumentation relies only on the CLR (runtime) version, not the compile-time target. Even if an application is built for .NET Framework older than 4.6.1, as long as it runs on a supported CLR version, automatic instrumentation will work normally.
+
+Regarding custom instrumentation, NuGet package version 2.0 and above will no longer include binaries for versions of .NET Framework older than 4.6.1. The NuGet package will only work correctly with applications that target (at compile time):
+
+- .NET Framework 4.6.1 or above
+- .NET Core 3.1 or above
+- .NET 5 or above
+- .NET Standard 2.0 or above (includes .NET Core 2.0 and above)
+
+### Rollout strategy when using automatic and custom instrumentation
+
+If your application uses both automatic and custom instrumentation, your application traces may be disconnected until both components are upgraded to v2.x. Follow the steps below to upgrade the components in the recommended order.
+
+**Note:** To minimize application overhead, align the versions of the automatic and custom instrumentation. Using two different versions of the 2.x .NET Tracer will still result in complete traces but it requires additional overhead, so Datadog recommends minimizing the amount of time that the versions are mismatched.
+
+#### .NET Core and .NET 5+
+
+##### If you were using v1.28.8 or greater
+
+On .NET Core, to avoid disconnected traces during the upgrade process:
+
+1. Upgrade the nuget package to the latest version of the tracer
+2. Upgrade the automatic instrumentation package to the matching version of the tracer
+
+Indeed, on .NET Core, if the assembly version from the nuget package is newer, the CLR can load that one instead of the assembly from the tracer's home folder.
+
+##### If you were using v1.28.7 or lower
+
+In that case, the migration will create disconnected traces one last time. **You should update the Tracer Home (ie serverside) first though**, then the nuget.
```

Comment:
Really? 🤔 So for <1.28.7 they should do MSI then NuGet, but for >=1.28.8, they do NuGet first? What's the reason for that? Did we relax the netcore loading conditions in 1.28.8? What happens if they do NuGet first in that situation?

```suggestion
In that case, the migration will create disconnected traces one last time. **You should update the Tracer Home (ie serverside) first though**, then the NuGet.
```

Reply:
I don't recall the details we had discussed at the time. The conclusion was that upgrading the NuGet first would lead to missing traces as IIRC there were breaking changes before that (CallTarget vs CallSite maybe, sorry I really don't remember). 
So upgrading the NuGet first would make us enter that scenario because of .NetCore loading conditions. Thus why we suggest starting with the MSI as disconnected traces was better that missing ones.

cc @lucaspimentel and @kevingosse cause the only thing I recall is that we have discussed that together :)

Reply:
> the only thing I recall is that we have discussed that together

Then you remember more than I do 😅

Reply:
Ok I have done a bit of archeology in the comments of the file. @lucaspimentel mentionned that previous to version 1.28.6 we could face crashed as - in this version - he has merged multiple DLLs into one to avoid making functions calls across assemblies that could potentially have internal breaking changes. [https://github.com/DataDog/dd-trace-dotnet/pull/1443](https://www.google.com/url?q=https://github.com/DataDog/dd-trace-dotnet/pull/1443&sa=D&source=docs&ust=1644836181552487&usg=AOvVaw0U9HIz4Vs82FDcc3yLZCBp)

Reply:
And to close the loop on this one, I've done the test that I should have done when the topic has been raised and indeed with a `1.28.4` msi and a 2.x nuget, we lose automatic spans, we just get custom ones (without any error in logs btw which seems weird).

Reply:
> we lose automatic spans, we just get custom ones (without any error in logs btw which seems weird).

Interesting... I definitely expected logs because the 2.x Datadog.Trace will get loaded at startup and then PInvoke back to the profiler to load all of the integrations, and at this point the PInvoke should fail because the older `1.28.4` doesn't have the methods